Mathlib/RingTheory/AdjoinRoot.lean

```diff
@@ -266,43 +269,81 @@ theorem of.injective_of_degree_ne_zero [IsDomain R] (hf : f.degree ≠ 0) :
     rwa [degree_C h_contra, zero_le_degree_iff]
 #align adjoin_root.of.injective_of_degree_ne_zero AdjoinRoot.of.injective_of_degree_ne_zero
 
-variable [CommRing S]
+/-- `commutativity` is a tactic that discharges goals of the form `Commute a b`
+in commutative semirings by using `aesop`.
 
-/-- Lift a ring homomorphism `i : R →+* S` to `AdjoinRoot f →+* S`. -/
-def lift (i : R →+* S) (x : S) (h : f.eval₂ i x = 0) : AdjoinRoot f →+* S := by
-  apply Ideal.Quotient.lift _ (eval₂RingHom i x)
+It is currently only used in `AdjoinRoot.lift`. -/
+macro "commutativity" : tactic =>
+  `(tactic| first | aesop |
```

Comment:
This should probably raise an error if there is no `Commute` in the goal

Reply:
@eric-wieser Do you know how to check this? I don't...

Reply:
Are Aesop rulesets a thing? If not, using `guard_target` would probably work.

Reply:
`guard_target` isn't quite what we want here: for example, `lift` wants to use `commutativity` to solve `∀ r, Commute (i r) x`.

WRT rulesets: do you mean we should set the `commutativity` tactic to use a commutativity ruleset exclusively? I feel like that's premature optimization.

Reply:
Yes, I think we should do that because it matches the pattern set by `measurability`. I don't think that `commutativity` as a tactic should be solving everything that `aesop` can solve.

Reply:
@eric-wieser @Vierkantor do you want me to try copying the measurability tactic file and build a commutativity tactic along its line?